src/manifolds/Euclidean.jl

```diff
@@ -46,183 +29,232 @@ the [`Euclidean`](@ref) manifold itself, or the [`Sphere`](@ref), where every
 tangent space (as a plane in the embedding) uses this metric (in the embedding).
 
 Since the metric is independent of the field type, this metric is also used for
-the Hermitian metrics, i.e. metrics that are analogous to the Euclidean but where
-the field type of the manifold is `ℂ`.
+the Hermitian metrics, i.e. metrics that are analogous to the `EuclideanMetric`
```

Comment:
Perhaps it's clearer to say that the metric defines an inner product equivalent to the Frobenius inner product, implemented by `dot`, and give the equation (and its interpretation as complexifying. Then manifolds that use this metric can reference this docstring instead of re-explaining it. This also generalizes to other fields. e.g. quaternions.

Reply:
Also, since `Euclidean` supports multidimensional arrays, it would be more accurate to define the Frobenius norm for `Euclidean` equivalently as the element-wise sum of the Hadamard product of the conjugate:

`g_x(v,w) = \sum_{i,\dots,j} \overline{v}_{i \dots j} w_{i \dots j}`

instead of using the Hermitian transpose, which presumes no more than 2 dimensions.

I think more accurately, `dot` implements a Hilbert-Schmidt operator inner product, since any numeric type is free to overload `conj` (it doesn't need to be complex conjugation). e.g. Quaternions.jl uses `conj` for quaternionic conjugation, and `dot` applied to arrays of quaternions still works. Manifolds on non-real/non-complex manifolds will probably be rare, so I don't think we need to document this.